src/nunit.analyzers/BaseAssertionAnalyzer.cs

```diff
@@ -16,24 +23,19 @@ public override void Initialize(AnalysisContext context)
 
         protected abstract void AnalyzeAssertInvocation(OperationAnalysisContext context, IInvocationOperation assertOperation);
 
-        protected static bool IsAssert(IOperation operation)
-        {
-            return operation is IExpressionStatementOperation expressionOperation &&
-                expressionOperation.Operation is IInvocationOperation invocationOperation &&
-                IsAssert(invocationOperation);
-        }
-
-        protected static bool IsAssert(IInvocationOperation invocationOperation)
+        protected bool IsAssert(IInvocationOperation invocationOperation)
         {
-            return invocationOperation.TargetMethod.ContainingType.IsAssert();
+            return this.includeClassic
+                ? invocationOperation.TargetMethod.ContainingType.IsAnyAssert()
+                : invocationOperation.TargetMethod.ContainingType.IsAssert();
```

Comment:
Instead of conditional code in the base class, I would prefer this to be a virtual method which is overridden only in the ConstActualValueUsageAnalyzer. It is a condition set in a constructor which doesn't "smell" right.
If more analyzers would need this, we can do this override in a ClassicBaseAssertionAnalyzer instead.

Reply:
Yeah, in this situation I always have the "carry an extra field in the instance vs the extra overhead of a virtual method call" debate with myself, and I rarely end up with a strong opinion about it. Will amend shortly.